packages/typedoc-plugin-appium/lib/converter/converter.ts

```diff
@@ -0,0 +1,417 @@
+      this.#log.verbose('No {MethodMap} found in class %s', refl.name);
+      return routes;
+    }
+
+    const routeProps = filterChildrenByKind(methodMap, ReflectionKind.Property);
+
+    if (!routeProps.length) {
+      this.#log.warn('No routes found in {MethodMap} of class %s', refl.name);
+      return routes;
+    }
+
+    for (const routeProp of routeProps) {
+      const {originalName: route} = routeProp;
+
+      if (!isRoutePropDeclarationReflection(routeProp)) {
+        this.#log.warn('Empty route in %s.%s', refl.name, route);
+        continue;
+      }
+
+      const httpMethodProps = filterChildrenByGuard(routeProp, isHTTPMethodDeclarationReflection);
+
+      if (!httpMethodProps.length) {
+        this.#log.warn('No HTTP methods found in route %s.%s', refl.name, route);
+        continue;
+      }
+
+      for (const httpMethodProp of httpMethodProps) {
+        const {comment, originalName: httpMethod} = httpMethodProp;
+
+        const commandProp = findChildByNameAndGuard(
+          httpMethodProp,
+          NAME_COMMAND,
+          isCommandPropDeclarationReflection
+        );
+
+        // commandProp is optional.
+        if (!commandProp) {
+          continue;
+        }
+
+        if (!_.isString(commandProp.type.value) || _.isEmpty(commandProp.type.value)) {
+          this.#log.warn('Empty command name found in %s.%s.%s', refl.name, route, httpMethod);
+          continue;
+        }
+        const command = String(commandProp.type.value);
+
+        const payloadParamsProp = findChildByNameAndGuard(
+          httpMethodProp,
+          NAME_PAYLOAD_PARAMS,
+          isReflectionWithReflectedType
+        );
+        const requiredParams = this.#convertRequiredCommandParams(payloadParamsProp);
+        const optionalParams = this.#convertOptionalCommandParams(payloadParamsProp);
+
+        const commandMap: CommandMap = routes.get(route) ?? new Map();
+
+        commandMap.set(command, {
+          command,
+          requiredParams,
+          optionalParams,
+          httpMethod,
+          route,
+          comment,
+        });
+
+        routes.set(route, commandMap);
+      }
+    }
+
+    return routes;
+  }
+
+  /**
+   * Finds commands in all classes within a project or module
+   * @param parent - Project or module
+   * @returns Info about the commands in given `parent`
+   */
+  #convertModuleClasses(parent: ParentReflection) {
+    let routes: RouteMap = new Map();
+    let executeCommands: ExecCommandDataSet = new Set();
+
+    const classReflections = parent
+      .getChildrenByKind(ReflectionKind.Class)
+      .filter((child) =>
+        isReflectionWithReflectedType(child)
+      ) as DeclarationReflectionWithReflectedType[];
+
+    for (const classRefl of classReflections) {
+      this.#log.verbose('Converting class %s', classRefl.name);
+      const newMethodMap = this.#convertMethodMap(classRefl);
+
+      if (newMethodMap.size) {
+        routes = new Map([...routes, ...newMethodMap]);
```

Comment:
why should we reassign routes every time instead of adding new entries to it?

Reply:
doesn't really matter, but if we did that, there'd need to be a `for..of` loop here.

Reply:
for me it is easier to read when minimum things are reassigned. Also, the oneliner via [forEach](https://stackoverflow.com/questions/32000865/simplest-way-to-merge-es6-maps-sets) also looks good

Reply:
iirc our style was to avoid `forEach` in lieu of `for..of`.  but anyway: this whole thing is better as a `reduce`.